File: frontend/features/camera/domain/model/camera.ts
export interface Camera {
  id: string;
  name: string;
  ip_address: string;
  network_id: string | null;
  status: "ONLINE" | "OFFLINE" | "ERROR";
  rtsp_url: string | null;
  onvif_port: number;
  manufacturer: string | null;
  model: string | null;
  created_at: string;
}

export interface Network {
  id: string;
  name: string;
  subnet: string;
  description: string | null;
  is_active: boolean;
  created_at: string;
}

export interface DiscoveredCamera {
  ip_address: string;
  port: number;
  manufacturer: string | null;
  model: string | null;
  rtsp_url: string | null;
  onvif_address: string | null;
}

File: backend/app/domains/camera/domain/entity/camera.py
from dataclasses import dataclass, field
from datetime import UTC, datetime
from enum import StrEnum
from uuid import UUID, uuid4


class CameraStatus(StrEnum):
    ONLINE = "ONLINE"
    OFFLINE = "OFFLINE"
    ERROR = "ERROR"


@dataclass
class Camera:
    name: str
    ip_address: str
    network_id: UUID | None = None
    status: CameraStatus = CameraStatus.OFFLINE
    rtsp_url: str | None = None
    onvif_port: int = 80
    manufacturer: str | None = None
    model: str | None = None
    firmware_version: str | None = None
    mac_address: str | None = None
    id: UUID = field(default_factory=uuid4)
    created_at: datetime = field(default_factory=lambda: datetime.now(UTC))
    updated_at: datetime = field(default_factory=lambda: datetime.now(UTC))
